datafusion/expr/src/expr.rs

```diff
@@ -1389,6 +1390,201 @@ impl Expr {
             | Expr::Placeholder(..) => false,
         }
     }
+
+    pub fn hash_node(&self, hasher: &mut AHasher) {
+        match self {
+            Expr::Alias(Alias { relation, name, .. }) => {
+                hasher.write_u8(0);
```

Comment:
would it be easier to use [`std::mem::discriminant`](https://doc.rust-lang.org/std/mem/fn.discriminant.html)?